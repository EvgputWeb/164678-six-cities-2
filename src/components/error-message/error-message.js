import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import ActionCreator from '../../store/action-creator';
import {ERROR_MESSAGE_HIDE_TIMEOUT} from '../../constants';


const ErrorMessage = ({errorMessage, onClearErrorMessage}) => {

  if (!errorMessage) {
    return null;
  }

  setTimeout(()=>{
    onClearErrorMessage();
  }, ERROR_MESSAGE_HIDE_TIMEOUT);

  return (
    <p><b>{errorMessage}</b></p>
  );
};


ErrorMessage.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  onClearErrorMessage: PropTypes.func.isRequired
};

const mapStateToProps = (store) => ({
  errorMessage: store.errorMessage
});

const mapDispatchToProps = (dispatch) => ({
  onClearErrorMessage: () => {
    dispatch(ActionCreator.setErrorMessage(``));
  }
});

export {ErrorMessage};
export default connect(mapStateToProps, mapDispatchToProps)(ErrorMessage);
