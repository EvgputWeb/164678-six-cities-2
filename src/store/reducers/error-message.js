import ActionTypes from '../action-types';

const INITIAL_STATE = ``;

const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case ActionTypes.SET_ERROR_MESSAGE: return action.payload;
  }
  return state;
};

export {reducer as errorMessage};
