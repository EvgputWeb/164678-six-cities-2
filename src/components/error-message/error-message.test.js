import React from 'react';
import renderer from 'react-test-renderer';
import {ErrorMessage} from './error-message';


it(`ErrorMessage renders correctly`, () => {

  const errMsg = renderer
  .create(
      <ErrorMessage
        errorMessage={`Test error`}
        onClearErrorMessage={jest.fn()}
      />
  )
  .toJSON();

  expect(errMsg).toMatchSnapshot();
});
