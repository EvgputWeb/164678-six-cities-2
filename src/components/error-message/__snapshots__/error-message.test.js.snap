// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly 1`] = `
<p>
  <b>
    Test error
  </b>
</p>
`;
